src/test/java/net/snowflake/client/jdbc/ResultSetLatestIT.java

```diff
@@ -39,6 +40,44 @@ public ResultSetLatestIT() {
     super(queryResultFormat);
   }
 
+  /**
+   * Test that when closing of results is interrupted by Thread.Interrupt(), the memory is released
+   * safely before driver execution ends.
+   *
+   * @throws Throwable
+   */
+  @Test
+  public void testMemoryClearingAfterInterrupt() throws Throwable {
```

Comment:
This test injects a fake InterruptedException into the terminate() function. This will be caught by ResultSet and converted into a SQLException. When the SQLException is caught, the test asserts that all memory used by the result set chunks has been freed.